scripts/ufo_gsi_plots.py

```diff
@@ -187,36 +270,74 @@ def generate_figs(inpath, outpath):
                     'satellite': satellite,
                     'cycle': cycle,
                     'channels': channels,
-                    'outdir': './'}
+                    'outdir': outpath}
 
-        data_df = get_data_df(file, channels)
+        data_df, gsi_use_flag = get_data_df(file, channels)
         
         # Create Dataframe appropriate for plotting data
         plot_df = pd.DataFrame()
         
         # Get channel counts for bar graph
         ufo_obs_count = []
         ufo_obs_count_qc = []
+        ufo_obs_count_err = []
         gsi_obs_count = []
         gsi_obs_count_qc = []
+        gsi_obs_count_err = []
+        
 
-        for i in channels:
-            ufo = data_df[f'brightness_temperature_{i}@hofx'] + data_df[f'brightness_temperature_{i}@ObsBias']
-            gsi = data_df[f'brightness_temperature_{i}@GsiHofXBc']
+        for i, chan in enumerate(channels):
+            ufo = data_df[f'brightness_temperature_{chan}@hofx']
+            gsi = data_df[f'brightness_temperature_{chan}@GsiHofXBc']
+            ufo_oberr = data_df[f'brightness_temperature_{chan}@EffectiveError']
+            gsi_oberr = data_df[f'brightness_temperature_{chan}@GsiFinalObsError']
+            
+            # Index by EffectiveQC = 0
+            qc_df = data_df.loc[(data_df[f'brightness_temperature_{chan}@EffectiveQC'] == 0)]
 
-            qc_df = data_df.loc[data_df[f'brightness_temperature_{i}@EffectiveQC'] == 0]
+            qc_flag_ufo = qc_df[f'brightness_temperature_{chan}@hofx']
+            qc_flag_gsi = qc_df[f'brightness_temperature_{chan}@GsiHofXBc']
+            qc_flag_ufo_oberr = qc_df[f'brightness_temperature_{chan}@EffectiveError']
+            qc_flag_gsi_oberr = qc_df[f'brightness_temperature_{chan}@GsiFinalObsError']
 
-            qc_ufo = qc_df[f'brightness_temperature_{i}@hofx'] + qc_df[f'brightness_temperature_{i}@ObsBias']
-            qc_gsi = qc_df[f'brightness_temperature_{i}@GsiHofXBc']
-    
-            ufo_obs_count.append(len(ufo))
-            ufo_obs_count_qc.append(len(qc_ufo))
-            gsi_obs_count.append(len(gsi))
-            gsi_obs_count_qc.append(len(qc_gsi))
+            # If the gsi data was assimilated, index by GsiFinalObsError < 1e9
+            if gsi_use_flag[i] == 1:
+                
+                error_df = data_df.loc[data_df[f'brightness_temperature_{chan}@GsiFinalObsError'] < 1e9]
+
+                err_ufo = error_df[f'brightness_temperature_{chan}@hofx']
+                err_gsi = error_df[f'brightness_temperature_{chan}@GsiHofXBc']
+                err_ufo_oberr = error_df[f'brightness_temperature_{chan}@EffectiveError']
+                err_gsi_oberr = error_df[f'brightness_temperature_{chan}@GsiFinalObsError']
+
+            else:
```

Comment:
I don't know if this below here is quite right/the best. If !=1, we probably shouldn't use the qc_df, but rather, we should not use this at all. I think anyways.

Reply:
So just pass? Or somehow make everything NaN? I know where you're coming from, but wouldn't it just be all NaN's using the qc_df?

Reply:
If I understand the code correctly, just pass. As you are concatenating in the loop, right?

Reply:
I don't think I can pass, because then they wont be the same size in the end. I can apply nans to those four variables instead.